Error generating code from CDDL with optional array

When generating Code with `zcbor code -c pet.cddl -d -e -t Pet  --output-cmake ~/generated/pet` the optional element `error` causes the Error 
```
AssertionError: Must be a direct call to zcbor to guarantee that payload and elem_count"
are cleaned up after a failure.
```
If `error` is not optional code is generated sucessfully, and other elements can be marked as optional without throwing an error.
pet.cddl:
```
Pet = [
    name: [ +tstr ],
    ?error: arr,
]

arr = [
    dummy: uint,
]
```

--- a/zcbor/zcbor.py
+++ b/zcbor/zcbor.py
@@ -2589,9 +2589,6 @@ class CodeGenerator(CddlXcoder):
                     f"This code needs self.mode to be 'decode', not {self.mode}."
                 if not self.repeated_single_func_impl_condition():
                     decode_str = self.repeated_xcode(union_int)
-                    assert "zcbor_" in decode_str, \
-                        """Must be a direct call to zcbor to guarantee that payload and elem_count"
-are cleaned up after a failure."""
                     return f"({self.present_var_access()} = {self.repeated_xcode(union_int)}, 1)"
                 func, *arguments = self.repeated_single_func(ptr_result=True)
                 return (

